### PR TITLE
Cook now step and timer fragments

### PR DESCRIPTION
**Pull request by zcmcxm**

### CODE REVIEW

File: app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java
Comment (fmikov) [inline app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java:1]: Some variables/functions have an unnecessary (?) "m" in them, for example ismPaused(). Maybe the code would be clearer if they were removed, unless there is a specific reason for them.

File: app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        mStopTimeInFuture = getRealTime() + mMillisInFuture;
+        mHandler.sendMessage(mHandler.obtainMessage(MSG));
+        mCancelled = false;
+        mPaused = false;
+    }
+
+    /**
+     * Pause the countdown.
+     */
+    public long pause() {
```
Comment (fmikov) [inline app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java:99]: Could the pause() and resume() methods cause problems if the user clicks on the corresponding buttons multiple times, with a delay? Pausing once and then once again after 5 minutes would break the timer. Consider adding checks for such cases.

File: app/src/main/java/com/github/siela1915/bootcamp/TimerViewModel.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    public long resumeTimer(int index) {
+        if (timerList.containsKey(index)) {
+            CountDownTimerWithPause timer = timerList.get(index);
+            if (timer.ismPaused()) return timerList.get(index).resume();
+            return timer.getRemainingTime();
+        }
+        return 0;
+    }
+
+//    // TODO: test only
```
Comment (fmikov) [inline app/src/main/java/com/github/siela1915/bootcamp/TimerViewModel.java:28]: I think you forgot to remove this part if it won't be used

File: app/src/main/java/com/github/siela1915/bootcamp/CookNowTimerFragment.java
Comment (fmikov) [inline app/src/main/java/com/github/siela1915/bootcamp/CookNowTimerFragment.java:1]: The coverage for this file is kind of low, but overall the coverage is okay. If you find the time for it, consider writing a few more tests for this class

**Review by fmikov — APPROVED** [review]
Overall the code looks good, there we're some edge cases for the timer pausing/unpausing and naming conventions that you could check though.

File: app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java (hunk 1 above)
Comment (zcmcxm) [reply app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java:99]: Thanks for reminding! :D I have the checking logic implemented in the TimerViewModel because I think how those edge cases are handled is open for the component that uses this timer to decide. Let me know if you have further concern on this. :)

Comment (fmikov) [reply app/src/main/java/com/github/siela1915/bootcamp/CountDownTimerWithPause.java:99]: Oh I missed that my bad. Looks good then!
